tests/dialogBoxes.spec.ts

```diff
@@ -0,0 +1,33 @@
+import { test, expect } from "@playwright/test";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto("/");
+  await page.getByTitle("pettypes").click();
+  await expect(page.locator("h2")).toHaveText("Pet Types");
+});
+
+test.describe("Dialog Boxes", async () => {
+  test("Add and delete pet type", async ({ page }) => {
+    await page.getByRole("button", { name: "Add" }).click();
+
+    const nameInputField = page.locator("#name");
+    await expect(nameInputField).toBeVisible();
+
+    await nameInputField.fill("pig");
+    await page.getByRole("button", { name: "Save" }).click();
+
+    await expect(page.locator("table tr td input").last()).toHaveValue("pig");
+
+    page.on("dialog", (dialog) => {
+      expect(dialog.message()).toEqual("Delete the pet type?");
+      dialog.accept();
+    });
+
+    await page
+      .locator("table tr")
+      .last()
+      .getByRole("button", { name: "Delete" })
+      .click();
+    await expect(page.locator("table tr").last()).not.toHaveText("pig");
```

Comment:
This assertion will always be "false positive" because table does not have "text" it has "values". The validation that you have used in line 19. So the table will never have text "pig", even if you add this record to the table.

Also, look into the locators at lines 19, 27, and 31. Do you see similarities? How can you optimize them to reuse the locator?

Reply:
Got it, thanks!

I’ve created a constant for the repeated code lines.